Email Service/EmailSender.cls

```diff
@@ -0,0 +1,23 @@
+public class EmailSender {
+    public static void sendOpportunityUpdateMail(List<Opportunity> emailopportunityList)
+    {
+        List<Messaging.SingleEmailMessage> allmsg = new List<Messaging.SingleEmailMessage>();
+        EmailTemplate emailTemplateRecord = [SELECT Id FROM EmailTemplate WHERE DeveloperName LIKE 'OpportunityUpdateEmail' LIMIT 1];
+        List<Opportunity> opportunityList = [SELECT Id,Name,Owner.Email FROM Opportunity Where Id in:emailopportunityList];
```

Comment:
It seems like you only need the opportunity Ids here so it's better if this method accept List<Id>

Reply:
ok